### PR TITLE
RFC017: Initial commit

### PR DESCRIPTION
**Pull request by reinkrul**

### CODE REVIEW

File: rfc/rfc017-distributed-network-grpc-v2.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# RFC017 Distributed Network Protocol (v2) using gRPC
+
+|  |  |
+| :--- | :--- |
+| Nuts foundation | R.G. Krul |
+| Request for Comments: 017 |  |
+| Updates: RFC017 | December 2021 |
```
Comment (woutslakhorst) [inline rfc/rfc017-distributed-network-grpc-v2.md:7]: Replaces RFC005?

File: rfc/rfc017-distributed-network-grpc-v2.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## Distributed Network Protocol (v2) using gRPC
+
+### Abstract
+
+This RFC specifies a new gRPC-based distributed network protocol. It succeeds version 1 of the protocol as defined by [RFC005](rfc005-distributed-network-using-grpc.md).
+At first, it will provide additions/optimizations over version 1, when feature complete it will replace it.
```
Comment (woutslakhorst) [inline rfc/rfc017-distributed-network-grpc-v2.md:14]: describing its life path isn't needed. It simply replaces rfc005

File: rfc/rfc017-distributed-network-grpc-v2.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+* **peer**: remote Nuts node that communicates with the local node.
+* **node identity**: (a.k.a. node DID) the DID a node uses to identify itself on the Nuts network. Multiple logical nodes (e.g. a cluster) may share the same node identity.
+* **private transactions**: transactions that are meant for a specific receiver (node) on the network. The transaction payload is solely shared with that particular node.
+
+Other terminology comes from the [Nuts Start Architecture](rfc001-nuts-start-architecture.md#nuts-start-architecture).
+
+## X. Authenticating node identity
+
+Particular exchanges (private transactions) might require authentication of the peer's identity. This identity (a.k.a. node identity) is specified by [RFC015 Node identity](rfc015-node-identity.md).
+If a node has configured a node DID, it MUST send it as `nodeDID` gRPC header when establishing inbound or outbound connections.
```
Comment (woutslakhorst) [inline rfc/rfc017-distributed-network-grpc-v2.md:40]: configuration is an implementation choice. Also I would say MAY,(but that would limit functionality)

File: rfc/rfc017-distributed-network-grpc-v2.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+The node DID SHOULD be authenticated when received, so that exchanges that require authentication can proceed without interruption.
```
Comment (woutslakhorst) [inline rfc/rfc017-distributed-network-grpc-v2.md:41]: What happens if it is wrong? Disconnect?

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

File: rfc/rfc017-distributed-network-grpc-v2.md (hunk 4 above)
Comment (reinkrul) [reply rfc/rfc017-distributed-network-grpc-v2.md:41]: Wasn't sure about that, currently I'm implementing that it's logged and just proceeds. Maybe disconnecting is clear to the connecting peer?

Comment (woutslakhorst) [reply rfc/rfc017-distributed-network-grpc-v2.md:41]: The peer might assume everything was correct, so it needs some kind of feedback >-)

Comment (reinkrul) [reply rfc/rfc017-distributed-network-grpc-v2.md:41]: All in favor of disconnecting, say "aye".

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**